kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -0,0 +1,227 @@
+            .endMetadata()
+            .addNewRule()
+                .addToApiGroups("")
+                .addToVerbs("get", "create", "delete", "patch")
+                .addToResources("namespaces")
+            .endRule()
+            .addNewRule()
+                .addToApiGroups("apiextensions.k8s.io")
+                .addToVerbs("get", "list", "watch", "create", "delete", "patch", "delete")
+                .addToResources("customresourcedefinitions")
+                .endRule()
+            .build();
+    // @formatter:on
+
+    private final Path resourceDirectory;
+    private final List<PathMatcher> clusterRolePathMatchers;
+    private List<ClusterRole> clusterRoles;
+    private List<ClusterRoleBinding> roleBindings;
+
+    public LocallyRunningOperatorRbacHandler(Path resourceDirectory, String... clusterRoleFileGlobs) {
+        Objects.requireNonNull(resourceDirectory);
+        this.resourceDirectory = resourceDirectory;
+        clusterRolePathMatchers = Arrays.stream(clusterRoleFileGlobs).map(g -> FileSystems.getDefault().getPathMatcher("glob:**/" + g)).toList();
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        try (var adminClient = OperatorTestUtils.kubeClient();
+                var files = Files.list(resourceDirectory)) {
+            // The test framework itself needs these roles.
+            Stream<ClusterRole> frameworkClusterRoles = Stream.of(frameworkClusterRole);
+            clusterRoles = Stream.concat(loadClusterRoles(files, adminClient), frameworkClusterRoles).toList();
+            clusterRoles.forEach(r -> {
+                LOGGER.trace("Creating/patching: {}", name(r));
+                adminClient.resource(r).createOr(EditReplacePatchable::patch);
+            });
+
+            roleBindings = this.clusterRoles.stream().map(this::bindingForRole).toList();
+            roleBindings.forEach(roleBinding -> {
+                LOGGER.trace("Creating role binding: {}", name(roleBinding));
+                adminClient.resource(roleBinding).createOr(EditReplacePatchable::patch);
+            });
+
+            LOGGER.info("Applied Operator RBAC rules rewritten in terms of user {}.", impersonatedUser);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+    }
+
+    private ClusterRoleBinding bindingForRole(ClusterRole clusterRole) {
+        return new ClusterRoleBindingBuilder().withNewMetadata().withName(name(clusterRole) + "-" + impersonatedUser + "-binding").endMetadata()
+                .addNewSubject().withKind("User").withName(impersonatedUser).withApiGroup("rbac.authorization.k8s.io").endSubject()
+                .withNewRoleRef().withKind("ClusterRole").withName(name(clusterRole)).withApiGroup("rbac.authorization.k8s.io").endRoleRef().build();
+    }
+
+    private @NonNull Stream<ClusterRole> loadClusterRoles(Stream<Path> files, KubernetesClient adminClient) {
```

Comment:
Thinking about test performance/duration is it worth pulling this into a `beforeAll` ? As we don't expect the file system to change during test exectuion.

Reply:
I've reworked things so that the extension is a static field, but it still uses beforeEach afterEach to configure each test the way we want. `afterAll` is only used to close a client. This means we load the files once per IT, and create/destroy the roles/bindings per test.